gef-ui/server/controllers/provided-facility/index.js

```diff
@@ -13,9 +14,20 @@ const providedFacility = async (req, res) => {
 
   try {
     const { details } = await api.getFacility({ facilityId, userToken });
+    const deal = await api.getApplication({ dealId, userToken });
     const facilityTypeConst = FACILITY_TYPE[details.type.toUpperCase()];
     const facilityTypeString = facilityTypeConst ? facilityTypeConst.toLowerCase() : '';
 
+    let previousPage;
+    if (!isFacilityEndDateEnabledOnGefVersion(parseDealVersion(deal.version)) || details.isUsingFacilityEndDate === null) {
```

Comment:
Should verify ACs for this logic

Reply:
This could probably be it's own function to keep this function more controller-like? what do you think

Reply:
a next/previous page state machine you say?

Reply:
Vijay happy with this behaviour